src/languages/dos.js

```diff
@@ -141,25 +160,39 @@ export default function(hljs) {
       built_in: BUILT_INS
     },
     contains: [
+      COMMENT,
+      COMMENT_2,
       {
         className: 'variable',
         begin: /%%[^ ]|%[^ ]+?%|![^ ]+?!/
       },
       {
         className: 'function',
         begin: LABEL.begin,
-        end: 'goto:eof',
+        end: /\n/,
         contains: [
           hljs.inherit(hljs.TITLE_MODE, { begin: '([_a-zA-Z]\\w*\\.)*([_a-zA-Z]\\w*:)?[_a-zA-Z]\\w*' }),
-          COMMENT
+          COMMENT,
+          COMMENT_2
         ]
       },
+      {
+        className: 'function',
+        begin: DISK_CHANGE.begin,
+        end: /\n/,
+        contains: [ hljs.inherit(hljs.TITLE_MODE, { begin: '([_a-zA-Z]\\w*\\.)*([_a-zA-Z]\\w*:)?[_a-zA-Z]\\w*' }) ]
```

Comment:
Please explain what this is attempting to do.  Are you trying to highlight all path names?